mlir/include/mlir/Analysis/DataFlow/IntegerRangeAnalysis.h

```diff
@@ -83,6 +84,9 @@ LogicalResult staticallyNonNegative(DataFlowSolver &solver, Operation *op);
 /// to use a non-64-bit index.
 LogicalResult staticallyNonNegative(DataFlowSolver &solver, Value v);
 
+LogicalResult maybeReplaceWithConstant(DataFlowSolver &solver,
+                                       PatternRewriter &rewriter, Value value);
```

Comment:
This could be an `RewriterBase &` since this isn't necessarily being called in a pattern context?

Reply:
done